main/src/ca/uwaterloo/flix/language/phase/HtmlDocumentor.scala

```diff
@@ -0,0 +1,640 @@
+    docSection("Definitions", mod.defs.sortBy(_.sym.name), docDef)
+
+    sb.append("</body>")
+
+    sb.toString()
+  }
+
+  /**
+    * Generates the string representing the head of the HTML document.
+    */
+  private def mkHead(name: String): String = {
+    "<!doctype html><html lang='en'>" +
+      "<head>" +
+      "<meta charset='utf-8'/>" +
+      "<meta name='viewport' content='width=device-width,initial-scale=1'/>" +
+      "<link href='https://fonts.googleapis.com/css?family=Fira+Code&display=swap' rel='stylesheet'>" +
+      "<link href='https://fonts.googleapis.com/css?family=Oswald&display=swap' rel='stylesheet'>" +
+      "<link href='styles.css' rel='stylesheet'>" +
+      s"<title>Flix Doc | ${esc(name)}</title>" +
+      "</head>"
+  }
+
+  /**
+    * Documents a section, (Classes, Enums, Effects, etc.), containing a `group` of items, each being placed in a box.
+    *
+    * The result will be appended to the given `StringBuilder`, `sb`.
+    *
+    * If `group` is empty, nothing will be generated.
+    *
+    * @param name   The name of the section, e.g. "Classes".
+    * @param group  The list of items in the section, in the order that they should appear.
+    * @param docElt A function taking a single item from `group` and generating the corresponding HTML string.
+    *               Note that this will be already be wrapped in a box.
+    */
+  private def docSection[T](name: String, group: List[T], docElt: T => Unit)(implicit flix: Flix, sb: StringBuilder): Unit = {
+    if (group.isEmpty) {
+      return
+    }
+
+    sb.append("<section>")
+    sb.append(s"<h2>${esc(name)}</h2>")
+    for (e <- group) {
+      sb.append("<div class='box'>")
+      docElt(e)
+      sb.append("</div>")
+    }
+    sb.append("</section>")
+  }
+
+  /**
+    * Documents a collapsable subsection, (Signatures, Instances, etc.), containing a `group` of items.
+    *
+    * The result will be appended to the given `StringBuilder`, `sb`.
+    *
+    * If `group` is empty, nothing will be generated.
+    *
+    * @param name   The name of the subsection, e.g. "Signatures".
+    * @param group  The list of items in the section, in the order that they should appear.
+    * @param docElt A function taking a single item from `group` and generating the corresponding HTML string.
+    */
+  private def docSubSection[T](name: String, group: List[T], docElt: T => Unit)(implicit flix: Flix, sb: StringBuilder): Unit = {
+    if (group.isEmpty) {
+      return
+    }
+
+    sb.append("<details>")
+    sb.append(s"<summary><h3>${esc(name)}</h3></summary>")
+    for (e <- group) {
+      docElt(e)
+    }
+    sb.append("</details>")
+  }
+
+  /**
+    * Documents the given `Class`, `clazz`.
+    *
+    * The result will be appended to the given `StringBuilder`, `sb`.
+    */
+  private def docClass(clazz: Class)(implicit flix: Flix, sb: StringBuilder): Unit = {
+    docAnnotations(clazz.ann)
+    sb.append("<code>")
```

Comment:
We could use raw interpolated strings here (and still append the "big string" to the string builder). But I am not sure if it is better. Thoughts?

Reply:
I started out doing something similar, but it became pretty hard to read with the different indentation levels, so I changed it to this.

Reply:
Ok